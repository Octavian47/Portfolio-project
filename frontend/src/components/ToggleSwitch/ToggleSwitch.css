.toggle-switch-container {
    display: flex;
    align-items: center;
  }
  
  .toggle-label {
    margin-right: 12px; /* Increase space between label and switch */
    font-size: 16px; /* Increase font size */
    font-weight: bold; /* Make the label bold */
    color: #333; /* Adjust the color if needed */
  }
  
  .toggle-switch {
    position: relative;
    display: inline-block;
    width: 40px; /* Increase the width of the switch */
    height: 20px; /* Increase the height of the switch */
    vertical-align: middle;
  }
  
  .toggle-switch-checkbox {
    display: none;
  }
  
  .toggle-switch-label {
    display: block;
    overflow: hidden;
    cursor: pointer;
    border: 2px solid #ddd;
    border-radius: 20px;
    transition: background-color 0.2s ease-in;
    background-color: #ccc;
    width: 100%;
    height: 100%;
  }
  
  .toggle-switch-button {
    display: block;
    width: 18px; /* Increase the size of the toggle button */
    height: 18px;
    background: #fff;
    border-radius: 16px;
    transform: translateY(-50%);
    transition: left 0.2s ease-in;
    position: relative;
    top: 50%;
    left: 2px;
  }
  
  .toggle-switch-checkbox:checked + .toggle-switch-label .toggle-switch-button {
    left: 20px; /* Adjust the position for the checked state */
  }
  
  .toggle-switch-checkbox:checked + .toggle-switch-label {
    background-color: #2e8b57;
  }
  

  