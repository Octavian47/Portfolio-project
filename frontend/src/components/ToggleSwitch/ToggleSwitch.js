import React from 'react';
import './ToggleSwitch.css';

const ToggleSwitch = ({ isOn, handleToggle, id }) => {
  return (
    <div className="toggle-switch-container">
      <span className="toggle-label">Display in Portfolio</span>
      <span className="toggle-switch">
        <input
          checked={isOn}
          onChange={handleToggle}
          className="toggle-switch-checkbox"
          id={`toggle-switch-${id}`} // Unique ID for each switch
          type="checkbox"
        />
        <label
          className="toggle-switch-label"
          htmlFor={`toggle-switch-${id}`} // Bind label to the unique ID
        >
          <span className={`toggle-switch-button`} />
        </label>
      </span>
    </div>
  );
};

export default ToggleSwitch;
