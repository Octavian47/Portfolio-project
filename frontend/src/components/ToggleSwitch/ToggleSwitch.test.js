import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import ToggleSwitch from './ToggleSwitch';

describe('ToggleSwitch Component', () => {
  it('renders the ToggleSwitch with the correct label', () => {
    const { getByText } = render(
      <ToggleSwitch isOn={false} handleToggle={jest.fn()} id="test" />
    );

    expect(getByText('Display in Portfolio')).toBeInTheDocument();
  });

  it('renders the ToggleSwitch in the "off" state', () => {
    const { getByRole } = render(
      <ToggleSwitch isOn={false} handleToggle={jest.fn()} id="test" />
    );

    expect(getByRole('checkbox')).not.toBeChecked();
  });

  it('renders the ToggleSwitch in the "on" state', () => {
    const { getByRole } = render(
      <ToggleSwitch isOn={true} handleToggle={jest.fn()} id="test" />
    );

    expect(getByRole('checkbox')).toBeChecked();
  });

  it('calls the handleToggle function when clicked', () => {
    const handleToggle = jest.fn();
    const { getByRole } = render(
      <ToggleSwitch isOn={false} handleToggle={handleToggle} id="test" />
    );

    fireEvent.click(getByRole('checkbox'));

    expect(handleToggle).toHaveBeenCalledTimes(1);
  });

  it('assigns the correct ID to the input and label elements', () => {
    const { getByLabelText, container } = render(
      <ToggleSwitch isOn={false} handleToggle={jest.fn()} id="unique-id" />
    );

    // Check if the input element has the correct ID
    const input = container.querySelector('#toggle-switch-unique-id');
    expect(input).toBeInTheDocument();

    // Check if the label is associated with the correct input by ID
    const label = container.querySelector('label[for="toggle-switch-unique-id"]');
    expect(label).toBeInTheDocument();
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(
      <ToggleSwitch isOn={false} handleToggle={jest.fn()} id="test" />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
