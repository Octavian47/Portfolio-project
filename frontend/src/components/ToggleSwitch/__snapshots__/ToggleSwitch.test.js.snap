// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleSwitch Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="toggle-switch-container"
  >
    <span
      class="toggle-label"
    >
      Display in Portfolio
    </span>
    <span
      class="toggle-switch"
    >
      <input
        class="toggle-switch-checkbox"
        id="toggle-switch-test"
        type="checkbox"
      />
      <label
        class="toggle-switch-label"
        for="toggle-switch-test"
      >
        <span
          class="toggle-switch-button"
        />
      </label>
    </span>
  </div>
</DocumentFragment>
`;
